api/mesh/v1alpha1/gateway_route.proto

```diff
@@ -226,6 +226,13 @@ message MeshGatewayRoute {
         Redirect redirect = 3;
         Rewrite rewrite = 4;
         HeaderFilter response_header = 5;
+        // Option to indicate that during forwarding, the host header should be
+        // swapped with the hostname of the upstream host chosen by the Envoy's
+        // cluster manager.
+        // BE AWARE:
+        // - it's mutually exclusive with request_header filter which explicitly
+        //   replaces "host" header
+        bool auto_host_rewrite = 6;
```

Comment:
I'm not quite sure this is the right name, maybe something like `RewriteHostToBackendHostname`?

Reply:
Hmm, I'm not convinced to be honest. I think addition of word `backend` might introduce confusion. Having it as is maybe not be very explicit, but at least not create any potentially false assumptions. In both cases it would have to be explained (probably with examples) in docs.

Reply:
It's already called a `backend` in the `MeshGatewayRoute` object.

Reply:
@michaelbeaumont ok, I went with your suggestion